### DISCUSSION TITLE
Feature Request: Function to extract byte from blob

### DISCUSSION THREAD

**Discussion by matt-allan** [duckdb/duckdb#7071] [Ideas]
The feature:

Something similar to the [`get_byte`](https://www.postgresql.org/docs/9.0/functions-binarystring.html) function in Postgres. 

```sql
-- Get the byte at offset 0 ( \xAA / 170)
SELECT get_byte('\xAA\xAB\xAC'::BLOB, 0)
```

Use case:

I have millions of rows that I need to filter based on the first byte of the blob (the first byte is a discriminator for the type of the blob). I would like to be able to write a where clause along the lines of `WHERE account_id = ? AND get_byte(my_blob, 0) = ?`.

Alternatives:

- I can base64 encode the blob and then check the first character, but that requires reading the whole blob and encoding it
- I can register a UDF that checks the first byte but that requires copying the entire blob too
- I can parse out the first byte when I ingest the data (currently my best option)
- I can't use `array_extract` (`my_blob[0]`) because that doesn't work for blob types
- There's an [existing request](https://github.com/duckdb/duckdb/discussions/2114) to extend `LIKE` to work for blob types, but that's overkill for my needs

Implementation thoughts:

Would this make more sense as a new function, or should the slice syntax be extended to work for blobs? The latter seems more flexible to me.

**Comment by carlopi** [comment]
Independently of the feature requests, for your needs I think you can hack it together with two comparisons.
Say  `>= '\xD6'` combined with `< '\xD7'` should give only the blobs starting with `'\xD6'`.

**Comment by matt-allan** [comment]
I ended up writing a small C++ extension to handle this in the interim. I will hopefully be able to open source it soon.

**Comment by vi** [comment]
Maybe BLOBs should be castable to `UTINYINT[]`, so that bytes can be extracted?
